rotkehlchen/chain/arbitrum_one/modules/hyperliquid/decoder.py

```diff
@@ -0,0 +1,107 @@
+
+import logging
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.arbitrum_one.modules.hyperliquid.constants import (
+    BRIDGE_ADDRESS,
+    CPT_HYPER,
+    FINALIZE_WITHDRAWAL,
+)
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import DEFAULT_DECODING_OUTPUT
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.utils.misc import bytes_to_address
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.structures import DecoderContext, DecodingOutput
+    from rotkehlchen.chain.evm.structures import EvmTxReceiptLog
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+    from rotkehlchen.types import ChecksumEvmAddress, EvmTransaction
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class HyperliquidDecoder(DecoderInterface):
```

Comment:
IS Hyperliquid only in Arbitrum? And is it only deposit/withdrawals and nothing else? Does it need any balances class to detect staked/deposited/hidden balances?

Reply:
Yes it's only an arbitrum thing. In arbitrum there is only deposit withdrawals, think about it like an exchanges. They have their API to query balances https://hyperliquid.gitbook.io/hyperliquid-docs/for-developers/api/info-endpoint/spot#retrieve-a-users-token-balances and also you query the trades that you do also happen inside their product. https://hyperliquid.gitbook.io/hyperliquid-docs/for-developers/api/info-endpoint#retrieve-a-users-fills-by-time they have both SPOT and PERP but it is more famous for the PERPs

Reply:
So this PR only decodes the on chain deposit/withdrawals (since I need it for accounting) In another PR we can see to support the balances and maybe the history?